### ISSUE TITLE
Rethink "Featured Image" in the context of blocks 

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#27617]
While we're thinking through block-based themes and their templates, It might be worth expanding the concept of what a "Featured Image" is. 

In the past, the featured image was a single image, added and managed separately from the rest of the post content. For block-based sites however, that restriction is largely unnecessary. Rather than just an image, we should consider making it possible to set a video or audio embed as the "Featured" block. Also, since posts are made up of structured data, it's theoretically possible to signify any block from within post content as a "Featured" block. 

A couple possible use cases:

- A photo blog, where posts contain a single image. If no separate featured image exists, the first image in the post is used as the featured image.
- A podcast site, where each post's "featured" block is audio of the podcast episode.

**Comment by mtias** [comment]
Another thing to consider here are possible attributes of the featured media, like "focus point" if used as a background, or styles like "duotone" effect.

**Comment by annezazu** [comment]
[Pulling in feedback from the second round of FSE Outreach Program](https://make.wordpress.org/test/2021/02/18/fse-program-testing-call-2-build-a-homepage-with-site-editing-blocks/#comment-1399) testing that relates to re-imagining this as the current placeholder flow having people upload images is bit awkward:

> To me, I feel strange to be told to upload a featured image for each post here. I assume if each featured image are set, then this uploader won’t be shown. Still, I think it feels confusing. We should either: show some kind of placeholder to say this post has no image set as featured image; or, leave as is, but change the text to be more clear, and say no featured image is displayed because it is not set on the individual post. And that it will look better if you set them. Or notify about default image?

**Comment by aristath** [comment]
It's important to keep in mind that the featured image is not that important for the post itself... It's important for archives where the post is shown. A news site for example _needs_ featured images in articles/posts because that's what gets pulled to show the articles on the frontpage.

**Comment by beafialho** [comment]
> Rather than just an image, we should consider making it possible to set a video or audio embed as the "Featured" block.

Wanted to +1 here and give the example of a new theme we're designing where this would be a vital feature. Since this theme is aimed at videographers, it would contain galleries of videos and videos as featured images, displayed differently (as a full width background carousel, as a list of posts, mosaic grid, etc) as shown in the images below.

In this context, where a post would be a project, the featured image would be the core content. It would be useful as well to be able to add videos on the Query Loop, Latest Posts or Gallery block. Perhaps, if they were "featured videos" it would be more intuitive.

![Video Theme 1](https://user-images.githubusercontent.com/65220155/133093985-225a959f-df7d-4b8f-b05f-a3ee64d09022.jpg)
![Video Theme 5](https://user-images.githubusercontent.com/65220155/133094038-c84edb07-7459-4ece-801f-aa260196ce60.jpg)
![Post Page 3](https://user-images.githubusercontent.com/65220155/133094235-b32cea82-b976-4c1b-a4d7-68a634932c45.jpg)

**Comment by megphillips91** [comment]
### For journalism as a use case
 
There are quite a few content creators particularly those in journalistic roles who may have an established, efficient workflow that involves composing text only articles and placing just one image into the featured image. If that is my workflow, I am relying on the theme (in this case the page template) to do something with the featured image that does not require me to format the image before uploading it. The _goal in these workflows is speed_...and _whitespace to write well_ while not being bothered with page design or layout. 

So, as a site builder of this type of theme/platform, at the moment (5.9) I could disable the featured image, create a block pattern that visually achieves the same effect, and write some code that will set that first cover into the featured image in the background. Because for journalistic outcomes, we need to respect the data integrity which likely feeds interoperability within the rest api to news aggregators, search results, etc that specifically rely on a designated featured image (vs. video vs quote or whatever). 

### A solution

"Also, since posts are made up of structured data, it's theoretically possible to signify any block from within post content as a "Featured" block". I really love that idea, but I do not think the featured block can replace the featured image for journalism. However, if the theme author can constrain the 'feature this block' functionality to only certain blocks, this is an ideal solution.

![example for cover block](https://files.slack.com/files-pri/T024MFP4J-F02V61W0GRZ/image.png?pub_secret=38b9380e28)

**Comment by kjellr** [comment]
One more featured image enhancement suggestion, [as raised in a support forum thread](https://wordpress.org/support/topic/featured-image-height-3/): 

- The ability to add an external URL to use as a featured image.

**Comment by jasmussen** [comment]
Here's a mockup showing how featured image could become more of a data source than a piece of media. Just select "Featured Image" in the "Replace" dropdown:

<img width="3440" alt="Set Media   Text as featured image" src="https://user-images.githubusercontent.com/1204802/153553760-b8428925-8c84-431e-b8d9-59ebae66a47d.png">

The descriptive text in the dropdown could be improved. We might even add separation between media and "Data sources" if we find it necessary to add additional data sources, such as "First attached image" or something in that vein.

If no featured image is set, you could get the same appearance in the placeholder as you get when you insert the Featured Image block on its own:

<img width="1760" alt="When no featured image is set" src="https://user-images.githubusercontent.com/1204802/153553883-3c7d1657-c1fe-421e-84c0-51979e93d998.png">

**Comment by cbirdsong** [comment]
I like how that looks, though I think it would be best if this system was able to accommodate more than just the featured image. I would love to easily be able to pull any kind of post/site metadata into a block. (see #37753)

**Comment by jasmussen** [comment]
Nice, I did consider other data sources as I as exploring, though didn't share as it didn't feel intuitive enough:
<img width="337" alt="Screenshot 2022-02-17 at 08 44 12" src="https://user-images.githubusercontent.com/1204802/154428896-3a5ce6cc-4c3b-4fb7-8751-edccf8d53552.png">

The [6.0 preliminary road map suggests](https://make.wordpress.org/core/2022/01/26/preliminary-roadmap-for-6-0/):

> Explore the viability of inline tokens. This has come up repeatedly in the context of rendering dynamic strings (such as current date) in rich text blocks.

Do you think inline tokens (see also #21932) might play a role here? Ostensibly the dropdown shown above could contain other tokens for insertion as the URL.

**Comment by draganescu** [comment]
@jasmussen wouldn't a block transform from image to feature image do better than placing the option in the replace dropdown? It's more hidden but also more consistent IMO.

**Comment by jasmussen** [comment]
The problem is that would require changes to the block itself. The idea of the mockup here is that "featured image" is just a URL token, rather than a block.

**Comment by Sirjazzfeetz** [comment]
To summarize a few points from above:

- The ability to set any media block, or perhaps any block from post content, as "featured".

-------------------------

### Featured!.. where & how?

There are several posts about 'Featured Image' being a confusing feature;

..contextual explanation has been suggested ( [#30904](https://github.com/WordPress/gutenberg/issues/30904) ), but inconsistent use of the feature due to lack of functional knowledge (as someone mentioned), isn't helped with the schism between post-content and sidebar-content (as another mentioned). WYSIWYG improvements are welcome, but it's worth bearing in mind that these are limited to page view; complexity compounds when a 'featured image' is used on other pages, perhaps in many places. The "strange" (as one person put it) command to a "set featured image" for each and every post can create confusion;

> It's important to keep in mind that the featured image is not that important for the post itself... It's important for archives where the post is shown. A news site for example needs featured images in articles/posts because that's what gets pulled to show the articles on the frontpage.

### Multimedia Support

in regards to the initial request;

> It would be useful as well to be able to add videos on the Query Loop, Latest Posts or Gallery block

and another multimedia request;

> Allow the use of video stills or generated thumbnails of pdfs, etc. media support.

----------------------

### Updates to Existing Feature

Several updates have been suggested specific to the existing 'Featured Image' feature - including features lost from classic transition?.. such as the ability to display image caption / credit ( [#40946](https://github.com/WordPress/gutenberg/issues/40946) ) ( [#23408](https://github.com/Automattic/jetpack/issues/23408) ). Requests to expand it's support to other blocks ( [#35221](https://github.com/WordPress/gutenberg/issues/35221) ) are perhaps better implemented through the Block Bindings API ( [#37753](https://github.com/WordPress/gutenberg/issues/37753) ).

A popular visual example for Gutenberg integration being:

![Image](https://github.com/WordPress/gutenberg/assets/118427845/1957d060-dad6-474b-89ac-469442039f3a)
![Image](https://github.com/WordPress/gutenberg/assets/118427845/55309fd3-6f68-4e47-a59a-63cf12dd3efd)

----------------------

Beyond multimedia, designs today mix multiple media and have multi-layered editable images, perhaps any section or collection of elements could be "featured" to generate an "image"? (it could be more than that).

**Comment by colorful-tones** [comment]
Hi folks,
We are only one week away from the Beta 1 cut-off date for [WordPress 6.6](https://make.wordpress.org/core/6-6/). This issue hasn’t seen any movement in a while, so we (the editor triage leads of the 6.6 release) have decided to remove it from the [WordPress 6.6 Editor Tasks project board](https://github.com/orgs/WordPress/projects/179/views/1).

**Comment by jasmussen** [comment]
To revisit this, an opportunity is presenting itself. Here's the Image block toolbar:

![Image](https://github.com/user-attachments/assets/f79bbc46-74ae-4454-b848-966284986498)

Here's the Cover block toolbar:

![Image](https://github.com/user-attachments/assets/fbeb591d-aa9b-46df-8bc0-3cb2ddc18e10)

Here's Site Logo:

![Image](https://github.com/user-attachments/assets/c06c0958-d95c-4bac-83e0-9ed52d342024)

Here's Media & Text:

![Image](https://github.com/user-attachments/assets/a90df9bc-8e05-48eb-9ee7-12d0fd5c8c5e)

All of those have a "Replace" dropdown menu in the toolbar. They are all about to have a "Reset" option there as well, and in their setup state (before you add an image), they're about to have an "Add media|image|logo" option. _Some of those_ have "Use featured image". We just discussed moving these options into the media upload component itself (CC: @mamaduka). I wonder: can we simply add "Use featured image" to all of those, in the same spot?

**Comment by Mamaduka** [comment]
The option is already there. See #41476.

> can we simply add "Use featured image" to all of those, in the same spot?

Displaying a featured image requires additional handling in the block edit component and its save function. The latter will need a PHP renderer.

**Comment by jasmussen** [comment]
Site Logo makes less sense. It's not there for Image block, which in past conversations revolved around there already being a separate "Featured Image" block. But I wouldn't mind that still existing as a neat template shortcut, but the Image block getting that option too; even now you can add a block binding to replace the image source, it seems reasonable to then also just let you bind it to the featured image in a convenient spot. 

Coming back to this issue, it seems like the remaining issues revolve around the quality of life features we have for things like Cover (focal point, duotone, etc). Like this replace dropdown, this speaks to convergence across the media blocks; does Image need a focal point picker? One would certainly be useful when applying an aspect ratio. That might also intuitively offer a UI for it on the Featured Image block, which already has duotone. 

Would a mockup consolidating the missing features across the media blocks be a reasonable way to unblock this and bring it home? CC: @WordPress/gutenberg-design

**Comment by cbirdsong** [comment]
How would choosing a focal point work if the source is dynamic? That seems very odd.

**Comment by richtabor** [comment]
I am increasingly mindful of keeping blocks (an associated experiences) simple. 

Feels a tad complicated to essentially transform the block to a featured image—without a bi-directional behavior. Which then complicates both blocks a bit more :)

**Comment by jasmussen** [comment]
Just to clarify, you can already make the Image block dynamic using block bindings, like in this example:

https://github.com/user-attachments/assets/f192ed7a-ab68-4f87-b439-09274c2e7a5c

If you add to that an aspect ratio, say Square, that will be maintained even when the URL gets updated. The motivation for then adding focal point when an aspect ratio is applied, is that the dynamic source could potentially be portrait shape. And then, maybe, you'd like to ensure you always see photos from the top/center. Same motivation for needing a focal point on the Featured Image block.

I'd be happy to keep this simple, however. Does that mean we can close this issue?

**Comment by creativecoder** [comment]
> Does that mean we can close this issue?

I don't think this issue should be closed yet. The issue description raises the general concept of featured content, not just the featured image or image blocks.

For example, we're currently missing the ability to use a video or audio recording as the featured content for a post--I think adding that would be a good step toward resolving the issue.

**Comment by richtabor** [comment]
> For example, we're currently missing the ability to use a video or audio recording as the featured content for a post--I think adding that would be a good step toward resolving the issue.

There's not a concept of featured audio or featured video today, is there?

**Comment by richtabor** [comment]
> the first image in the post is used as the featured image.

This has happened too.

**Comment by creativecoder** [comment]
> There's not a concept of featured audio or featured video today, is there?

There are some plugins available, but nothing in core, afaik.

I think one of the things to explore is how to add featured content, and what kind of integration with the existing featured image is useful. Featured image (or post_thumbnail) is stored as post meta, referencing an attachment post. Would we do the same for featured audio and video files? Or expand further and allow promoting content from a specified block or blocks from a post? How would the theme define this? How would the user customize it, when desired?

For example, I could see a cover block from a single post being the featured content for that post, and the same cover block being used for that post in an archive template. How would we mark that as featured in the post and how would it be retrieved for the archive template? Do block bindings fit into this, at all?

**Comment by jasmussen** [comment]
Twenty Twenty-Five ships with a big featured image placeholder in the Page template. A new feature to "Show template" has also been added to the preview dropdown, which shows this in the regular page editor, and a followup PR explores making that choice persistent. Beyond the audio use case, it's possible I'm still missing some nuance as fara as moving forward on this one. After all, the featured image is used as opengraph thumbnail data in context of social sharing, I don't know that video or audio could ever fill that gap.

**Comment by iamtakashi** [comment]
It looks like this has been stalled, but I've heard people want to add a video as a featured image, and that would be nice if it support natively.

**Comment by BlackStar1991** [comment]
Is it possible to resume discussion of this proposal in 2026? For WP version 7.0
